static/src/js/pages/KeywordManagerPage/KeywordManagerPage.jsx

```diff
@@ -300,6 +369,57 @@ const KeywordManagerPage = () => {
         message="You are now viewing the live published keyword version. Changes made to this version will show up on the website right away."
         actions={warningModalActions}
       />
+      <CustomModal
+        show={showPublishModal}
+        toggleModal={() => setShowPublishModal(false)}
+        header="Publish New Keyword Version"
+        message={
+          (
+            <>
+              <Form.Group>
+                <Form.Label htmlFor="newKeywordVersion">Version Name:</Form.Label>
+                <Form.Control
+                  id="newKeywordVersion"
+                  type="text"
+                  value={newVersionName}
+                  onChange={(e) => setNewVersionName(e.target.value)}
+                  placeholder="Enter version"
+                />
+              </Form.Group>
+              {renderPublishStatus()}
+            </>
+          )
+        }
+        actions={
+          [
+            {
+              label: 'Cancel',
+              variant: 'secondary',
+              onClick: () => setShowPublishModal(false)
+            },
+            {
+              label: 'Publish',
+              variant: 'primary',
+              onClick: handlePublishVersion
+            }
+          ]
+        }
+      />
+
+      <CustomModal
+        show={showPublishingModal}
+        header="Publishing New Version"
+        message={
+          (
+            <div className="text-center">
+              <Spinner animation="border" role="status" className="mb-2" />
+              <p>Publishing...</p>
```

Comment:
It isn't instantaneous,, so I think maybe reword too "Publishing...  Please wait..."

Reply:
Done